consul/server_manager/server_manager_internal_test.go

```diff
@@ -125,17 +148,121 @@ func TestServerManagerInternal_New(t *testing.T) {
 	}
 }
 
-// func (sc *serverConfig) refreshServerRebalanceTimer(timer *time.Timer) {
-func TestServerManagerInternal_refreshServerRebalanceTimer(t *testing.T) {
-	sm := testServerManager()
+// func (sm *ServerManager) reconcileServerList(sc *serverConfig) bool {
+func TestServerManagerInternal_reconcileServerList(t *testing.T) {
+	tests := []int{0, 1, 2, 3, 4, 5, 10, 100}
+	for _, n := range tests {
+		ok, err := test_reconcileServerList(n)
+		if !ok {
+			t.Errorf("Expected %d to pass: %v", n, err)
+		}
+	}
+}
+
+func test_reconcileServerList(maxServers int) (bool, error) {
+	// Build a server list, reconcile, verify the missing servers are
+	// missing, the added have been added, and the original server is
+	// present.
+	const failPct = 0.5
+	sm := testServerManagerFailProb(failPct)
+	const numShuffleTests = 100
+	const uniquePassRate = 0.5
+
+	var failedServers, healthyServers []*server_details.ServerDetails
+	for i := 0; i < maxServers; i++ {
+		nodeName := fmt.Sprintf("s%02d", i)
+
+		node := &server_details.ServerDetails{Name: nodeName}
+		// Add 66% of servers to ServerManager
+		if rand.Float64() > 0.33 {
+			sm.AddServer(node)
+
+			// Of healthy servers, (ab)use connPoolPinger to
```

Comment:
I started with a fixed table and realized that w/ at least 3x server sizes (0,1,2) server sizes and >3 different variations of outcomes, it was easier to just cast a wide net and let probability do its thing.